### PR TITLE
fix: Don't cast all types to string for CSV

### PR DESCRIPTION
**Pull request by candiduslynx**
Fixes https://github.com/cloudquery/cloudquery-issues/issues/1644

Requires https://github.com/apache/arrow/pull/41595 & bump to arrow v16.1.0.

Diff in files generated:
1. date32 & date64: `2021-01-02 00:00:00` is now `2021-01-02 00:00:00Z` (so this can also be considered a follow-up for https://github.com/cloudquery/filetypes/pull/494)
2. Lists are marshaled as lists instead of strings, so it required a deep conversion for proper support (`ValueStr` ignores null value set to `""`)

### CODE REVIEW

File: csv/write.go [hunk 1]
```diff
@@ -57,59 +57,103 @@ func convertSchema(sch *arrow.Schema) *arrow.Schema {
...
 
-func isTypeSupported(t arrow.DataType) bool {
-	// list from arrow/csv/common.go
-	switch t.(type) {
+func convertType(dt arrow.DataType) arrow.DataType {
+	if typeSupported(dt) {
+		return dt
+	}
+	switch dt := dt.(type) {
+	case *arrow.MapType:
```
Comment (candiduslynx) [inline csv/write.go:74]: maps aren't supported

File: csv/write.go [hunk 2]
```diff
@@ -57,59 +57,103 @@ func convertSchema(sch *arrow.Schema) *arrow.Schema {
...
+	case *arrow.FixedSizeListType:
```
Comment (candiduslynx) [inline csv/write.go:75]: lists are converted as list(converted elem)

**Review by disq — APPROVED** [review]

**Comment by erezrokah** [comment]
Can we wait for someone to complain about the CSV format before fixing this? We recently did a breaking change in all our destinations that are using this module due to https://github.com/cloudquery/filetypes/pull/494 (which was justified as it fixed some stuff in Parquet, JSON and CSV for timestamps). Not sure we should do another one only for CSV

**Comment by candiduslynx** [comment]
@erezrokah please consider merging this

**Comment by candiduslynx** [comment]
@erezrokah @hermanschaaf @yevgenypats 
Closing this PR to free the GitHub board.
If you would like to merge this instead tag me so that I can reopen.
cc: @disq as well

**Pull request closed without merging**